rust/src/encodings/dictionary.rs

```diff
@@ -66,8 +66,8 @@ impl<'a> DictionaryDecoder<'a> {
         }
     }
 
-    async fn decode_index<T: ArrowDictionaryKeyType + Sync + Send>(&self) -> Result<ArrayRef> {
-        let index_decoder = PlainDecoder::<T>::new(self.reader, self.position, self.length)?;
+    async fn decode_index<T: ArrowDictionaryKeyType + Sync + Send>(&self, index_type: &DataType) -> Result<ArrayRef> {
```

Comment:
Should we remove the type paraemter `T` ?

Reply:
Need it for downcast ?